code/modules/reagents/reagent_containers/patch.dm

```diff
@@ -13,6 +13,16 @@
 	..()
 	icon_state = "bandaid" // thanks inheritance
 
+/obj/item/weapon/reagent_containers/pill/patch/attack(mob/living/M, mob/user)
+	if(!canconsume(M, user))
```

Comment:
no, canconsume literally just checks if you can eat it

the rest of the shit for that is handled in the pill code